src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -138,11 +142,12 @@ func parseParams(r *http.Request) (models.RequestParams, *xhttp.ParseError) {
 	return params, nil
 }
 
-func parseDebugAndIterFlags(r *http.Request) (bool, bool) {
+func parseDebugAndLegacyFlags(r *http.Request) (bool, bool) {
```

Comment:
should these be two methods so that its easier to get rid of legacy later ?

Reply:
I was thinking of adding a third flag which would enable the split block approach to test that when that is added, might split it out then to have a debug parser and a "fallback" parser